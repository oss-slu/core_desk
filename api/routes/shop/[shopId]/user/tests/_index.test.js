import { describe, expect, it } from "vitest";
import request from "supertest";
import { app } from "#index";
import { gt } from "#gt";
import { prisma } from "#prisma";
import { tc } from "#setup";

describe("/shop/[shopId]/user", () => {
  describe("GET", () => {
    it("returns a list of users for global admins", async () => {
      const res = await request(app)
        .get(`/api/shop/${tc.shop.id}/user`)
        .set(...(await gt({ ga: true })))
        .send();

      expect(res.status).toBe(200);
      expect(res.body.users).toHaveLength(1);
      expect(res.body.users[0]).toMatchSnapshot({
        id: expect.any(String),
      });
    });

    it("returns a list of users for shop-level admins", async () => {
      const res = await request(app)
        .get(`/api/shop/${tc.shop.id}/user`)
        .set(...(await gt({ sat: "ADMIN" })))
        .send();

      expect(res.status).toBe(200);
      expect(res.body.users).toHaveLength(1);
      expect(res.body.users[0]).toMatchSnapshot({
        id: expect.any(String),
      });
    });

    it("returns a list of users for operators", async () => {
      const res = await request(app)
        .get(`/api/shop/${tc.shop.id}/user`)
        .set(...(await gt({ sat: "OPERATOR" })))
        .send();

      expect(res.status).toBe(200);
      expect(res.body.users).toHaveLength(1);
      expect(res.body.users[0]).toMatchSnapshot({
        id: expect.any(String),
      });
    });

    it("returns a list of users for group admins", async () => {
      const res = await request(app)
        .get(`/api/shop/${tc.shop.id}/user`)
        .set(...(await gt({ sat: "GROUP_ADMIN" })))
        .send();

      expect(res.status).toBe(200);
      expect(res.body.users).toHaveLength(1);
      expect(res.body.users[0]).toMatchSnapshot({
        id: expect.any(String),
      });
    });

    it("returns 403 when the user is not an admin", async () => {
      const res = await request(app)
        .get(`/api/shop/${tc.shop.id}/user`)
        .set(...(await gt()))
        .send();

      expect(res.status).toBe(403);
      expect(res.body).toEqual({ error: "Unauthorized" });
    });
  });
});
