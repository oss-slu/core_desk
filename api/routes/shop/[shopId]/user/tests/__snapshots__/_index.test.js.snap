// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`/shop/[shopId]/user > GET > returns a list of users for global admins 1`] = `
{
  "firstName": "TestFirstName",
  "id": Any<String>,
  "lastName": "TestLastName",
  "name": "TestFirstName TestLastName",
}
`;

exports[`/shop/[shopId]/user > GET > returns a list of users for group admins 1`] = `
{
  "firstName": "TestFirstName",
  "id": Any<String>,
  "lastName": "TestLastName",
  "name": "TestFirstName TestLastName",
}
`;

exports[`/shop/[shopId]/user > GET > returns a list of users for operators 1`] = `
{
  "firstName": "TestFirstName",
  "id": Any<String>,
  "lastName": "TestLastName",
  "name": "TestFirstName TestLastName",
}
`;

exports[`/shop/[shopId]/user > GET > returns a list of users for shop-level admins 1`] = `
{
  "firstName": "TestFirstName",
  "id": Any<String>,
  "lastName": "TestLastName",
  "name": "TestFirstName TestLastName",
}
`;
